rust/lance-linalg/src/distance/l2.rs

```diff
@@ -150,7 +150,7 @@ impl L2 for f16 {
 impl L2 for f32 {
     #[inline]
     fn l2(x: &[Self], y: &[Self]) -> f32 {
-        l2_scalar::<Self, Self, 32>(x, y)
+        l2_scalar::<Self, Self, 16>(x, y)
```

Comment:
How did you choose this value? Could this be overfitting for the CPU you are benchmarking on?

Reply:
I tested in both on m2 and an amd 5950x machine.  Also tried to generate code with https://rust.godbolt.org/z/xoWf47WqE, with `--target-cpu=cascadelake`.

so `16 = 512 / 8 bit / 4 (f32)`.

Reply:
Could you leave a comment there so we remember this?

Reply:
Sure. done.